ui/components/TracePage/FString.tsx

```diff
@@ -0,0 +1,63 @@
+interface FValue {
+  source: string;
+  value: any;
+  formatted: string;
+}
+
+export type FlattenedFStringPart = string | FValue;
+export type RawFStringPart = FlattenedFStringPart | { __fstring__: RawFStringPart[] };
+
+export function flattenFString(parts: RawFStringPart[]): FlattenedFStringPart[] {
+  return parts.flatMap(part => {
+    if (typeof part === "object") {
+      if ("__fstring__" in part) {
+        return flattenFString(part.__fstring__);
+      }
+      if (typeof part.value === "object" && "__fstring__" in part.value) {
+        return flattenFString(part.value.__fstring__ as RawFStringPart[]);
+      }
+    }
+    return [part];
+  });
+}
+
+export const FString = ({ parts }: { parts: FlattenedFStringPart[] }) => {
+  let oddValue = false;
+  return (
+    <span>
+      {parts.map((part, i) => {
+        if (typeof part === "string") {
+          return <span key={i}>{part}</span>;
+        }
+        const color = oddValue ? "lightBlue" : "indigo";
+        oddValue = !oddValue;
+        const inner = part.formatted;
+
+        // This commented code allows seeing the original nested construction,
+        // if the value isn't flattened.
+        // const inner =
+        //   typeof part.value === "object" && "__fstring__" in part.value ? (
+        //     <FString parts={part.value.__fstring__ as FStringPart[]} />
+        //   ) : (
+        //     part.formatted
+        //   );
```

Comment:
I think we should either add this as an option or remove the comment. I lean toward removing it